src/analysis/normed_space/star/multiplier.lean

```diff
@@ -0,0 +1,429 @@
+(central : ∀ x y : A, snd x * y = x * fst y)
+
+localized "notation `𝓜(` 𝕜 `, ` A `)` := double_centralizer 𝕜 A" in multiplier_algebra
+
+namespace double_centralizer
+
+section nontrivially_normed
+
+variables (𝕜 A : Type*) [nontrivially_normed_field 𝕜] [non_unital_normed_ring A]
+variables [normed_space 𝕜 A] [smul_comm_class 𝕜 A A] [is_scalar_tower 𝕜 A A]
+
+instance : inhabited 𝓜(𝕜, A) :=
+{ default := ⟨1, λ x y, rfl⟩ }
+
+/-!
+### Algebraic structure
+
+Because the multiplier algebra is defined as the algebra of double centralizers, there is a natural
+injection `double_centralizer.to_prod_mop : 𝓜(𝕜, A) → (A →L[𝕜] A) × (A →L[𝕜] A)ᵐᵒᵖ` defined by
+`λ a, (a.fst, mul_opposite.op a.snd)`. We use this map to pull back the the ring, module and
+algebra structure from `(A →L[𝕜] A) × (A →L[𝕜] A)ᵐᵒᵖ` to `𝓜(𝕜, A)`. -/
+
+variables {𝕜 A}
+
+lemma range_to_prod : set.range to_prod = {lr : (A →L[𝕜] A) × _ | ∀ x y, lr.2 x * y = x * lr.1 y} :=
+set.ext $ λ x, ⟨by {rintro ⟨a, rfl⟩, exact a.central}, λ hx, ⟨⟨x, hx⟩, rfl⟩⟩
+
+instance : has_add 𝓜(𝕜, A) :=
+{ add := λ a b,
+  { to_prod := a.to_prod + b.to_prod,
+    central := λ x y, show (a.snd + b.snd) x * y = x * (a.fst + b.fst) y,
+      by simp only [continuous_linear_map.add_apply, mul_add, add_mul, central] } }
+
+instance : has_zero 𝓜(𝕜, A) :=
+{ zero :=
+  { to_prod := 0,
+    central := λ x y, (zero_mul y).trans (mul_zero x).symm } }
+
+instance : has_neg 𝓜(𝕜, A) :=
+{ neg := λ a,
+  { to_prod := -a.to_prod,
+    central := λ x y, show -a.snd x * y = x * -a.fst y,
+      by simp only [continuous_linear_map.neg_apply, neg_mul, mul_neg, central] } }
+
+instance : has_sub 𝓜(𝕜, A) :=
+{ sub := λ a b,
+  { to_prod := a.to_prod - b.to_prod,
+    central := λ x y, show (a.snd - b.snd) x * y = x * (a.fst - b.fst) y,
+      by simp only [continuous_linear_map.sub_apply, sub_mul, mul_sub, central] } }
+
+section scalars
+
+variables {S : Type*} [monoid S] [distrib_mul_action S A] [smul_comm_class 𝕜 S A]
+  [has_continuous_const_smul S A] [is_scalar_tower S A A] [smul_comm_class S A A]
+
+instance : has_smul S 𝓜(𝕜, A) :=
+{ smul := λ s a,
+  { to_prod := s • a.to_prod,
+    central := λ x y, show (s • a.snd) x * y = x * (s • a.fst) y,
+      by simp only [continuous_linear_map.smul_apply, mul_smul_comm, smul_mul_assoc, central] } }
+
+@[simp] lemma smul_fst (k : 𝕜) (a : 𝓜(𝕜, A)) : (k • a).fst = k • a.fst := rfl
+@[simp] lemma smul_snd (k : 𝕜) (a : 𝓜(𝕜, A)) : (k • a).snd = k • a.snd := rfl
+
+end scalars
+
+instance : has_one 𝓜(𝕜, A) := ⟨⟨1, λ x y, rfl⟩⟩
+
+instance : has_mul 𝓜(𝕜, A) :=
+{ mul := λ a b,
+  { to_prod := (a.fst.comp b.fst, b.snd.comp a.snd),
+    central := λ x y, show b.snd (a.snd x) * y = x * (a.fst (b.fst y)),
+      by simp only [central] } }
+
+instance : has_nat_cast 𝓜(𝕜, A) :=
+  { nat_cast := λ n, ⟨n, λ x y, by simp only [←nat.smul_one_eq_coe, prod.smul_fst, prod.smul_snd,
+      prod.fst_one, prod.snd_one, smul_apply n 1, one_apply, mul_smul_comm, smul_mul_assoc]⟩ }
```

Comment:
I am _pretty sure_ there's a lemma for that,  but I cannot find it right now. If you cannot find it either, could you leave a TODO (addressed to me or @eric-wieser)?

Reply:
I found a slightly more canonical proof, so I decided not to add the TODO. I would be surprised if we had anything to make it much simpler than this, but perhaps this is what you meant.

Reply:
Not what I meant, but I'm happy with it.